From f80b344cd3ebd098ba6e63aa1ee97a6223ffe7f0 Mon Sep 17 00:00:00 2001
From: Leah Rowe <leah@libreboot.org>
Date: Sun, 5 Nov 2023 17:25:20 +0000
Subject: [PATCH 1/1] don't print errors if GRUB_ERR_FILE_NOT_FOUND

this is part two of the quest to kill the prefix
error message. after i disabled prefix-related
messages, it still printed "error: ." on screen.

Signed-off-by: Leah Rowe <leah@libreboot.org>
---
 grub-core/kern/err.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/grub-core/kern/err.c b/grub-core/kern/err.c
index 53c734de7..c9488b748 100644
--- a/grub-core/kern/err.c
+++ b/grub-core/kern/err.c
@@ -107,7 +107,8 @@ grub_print_error (void)
     {
       if (grub_errno != GRUB_ERR_NONE)
 	{
-	  grub_err_printf (_("error: %s.\n"), grub_errmsg);
+          if (grub_errno != GRUB_ERR_FILE_NOT_FOUND)
+	    grub_err_printf (_("error: %s.\n"), grub_errmsg);
 	  grub_err_printed_errors++;
 	}
     }
-- 
2.39.2

